app/controllers/pools_controller.rb

```diff
@@ -29,12 +29,52 @@ def edit
   end
 
   def update
-    @pool = Pool.find(params[:id])
-    new_params = pool_params
-    new_params[:active] = new_params.has_key?(:active) && new_params[:active] == "true"
-    unless @pool.update(new_params)
-      @errors = @pool.errors.full_messages
-      render status: :bad_request
+    if params[:id] && auth_admin?
+      # allow any valid change but concurrency unsafe
+      @pool = Pool.find(params[:id])
+
+      update_params = pool_params
+      update_params[:active] = update_params.has_key?(:active) &&
+                               update_params[:active] == "true"
+      if @pool.update(update_params)
+        @persisted = true
+      else
+        @errors = @pool.errors.full_messages
+        render status: :bad_request
+      end
+    else
+      # for non-admin allow only updating active state to prevent
+      # pools stealing and other mistakes
```

Comment:
I am not really sure this logic is even necessary here. See my comment about auth.